cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -348,6 +348,10 @@ protected void startUp() throws Exception {
         Transactions.invalidateQuietly(txClient, tx);
         throw t;
       }
+    } catch (LinkageError e) {
```

Comment:
1. I'll look into what the behavior if destroy uses a class that's not found.
2. I added it for spark, mapreduce, worker. Workflow was not plagued by this issue (but I added a test case for it.